app/migrations/gorm-migrate.go

```diff
@@ -4,29 +4,286 @@ import (
 	"encoding/json"
 	"log"
 	"os"
-	"star-pos/features/user"
-	userData "star-pos/features/user/repository"
+	"star-pos/app/databases"
+	cartModel "star-pos/features/cart/model"
+	cartProductModel "star-pos/features/cart_product/model"
+	categoriesModel "star-pos/features/categories/model"
+	discountModel "star-pos/features/discount/model"
+	outletModel "star-pos/features/outlet/model"
+	productModel "star-pos/features/product/model"
+	transactionModel "star-pos/features/transaction/model"
+	transactionDetailModel "star-pos/features/transaction_detail/model"
+	userModel "star-pos/features/user/model"
+	"time"
 
 	"gorm.io/gorm"
 )
 
-func InitMigration(db *gorm.DB) {
-	file, err := os.ReadFile("userdummy.json")
+func InitMigration() {
+	roll := databases.DB.Begin()
+
+	file, err := os.ReadFile("dummy/userdummy.json")
 	if err != nil {
 		log.Fatal("error reading file: ", err)
 	}
-	var users []user.UserCore
+	var users []userModel.User
 	err = json.Unmarshal(file, &users)
 	if err != nil {
 		log.Fatal("error unmarshalling JSON: ", err)
 	}
 
-	repositoryInterface := userData.New(db)
-
 	for _, data := range users {
-		err = repositoryInterface.Insert(data)
+		var existingUser userModel.User
+		result := databases.DB.Where("id = ?", data.ID).First(&existingUser)
+		if result.Error != nil {
+			if result.Error == gorm.ErrRecordNotFound {
+				tx := databases.DB.Create(&data)
+				if tx.Error != nil {
+					roll.Rollback()
+					log.Fatal("error insert data: ", tx.Error)
+				}
+			}
+		} else {
+			log.Println("table already exist")
+			break
+		}
+	}
+
+	file, err = os.ReadFile("dummy/outletdummy.json")
+	if err != nil {
+		log.Fatal("error reading file: ", err)
+	}
+	var outlet []outletModel.Outlet
+	err = json.Unmarshal(file, &outlet)
+	if err != nil {
+		log.Fatal("error unmarshalling JSON: ", err)
+	}
+
+	for i, data := range outlet {
+		var existingOutlet outletModel.Outlet
+		result := databases.DB.Where("id = ?", data.ID).First(&existingOutlet)
```

Comment:
follow this suggestion for all tables https://github.com/ezabintangr/star-pos/pull/14#discussion_r1726889873